src/config.rs

```diff
@@ -0,0 +1,128 @@
+#[derive(PartialEq, Eq, Debug, Deserialize, Serialize, Clone)]
+#[serde(default)]
+pub struct Config {
+    /// The regions for DERP to use.
+    pub derp_regions: Vec<DerpRegion>,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            derp_regions: vec![default_derp_region()],
+        }
+    }
+}
+
+impl Config {
+    /// Make a config using a default, files, environment variables, and commandline flags.
+    ///
+    /// Later items in the *file_paths* slice will have a higher priority than earlier ones.
+    ///
+    /// Environment variables are expected to start with the *env_prefix*. Nested fields can be
+    /// accessed using `.`, if your environment allows env vars with `.`
+    ///
+    /// Note: For the metrics configuration env vars, it is recommended to use the metrics
+    /// specific prefix `IROH_METRICS` to set a field in the metrics config. You can use the
+    /// above dot notation to set a metrics field, eg, `IROH_CONFIG_METRICS.SERVICE_NAME`, but
+    /// only if your environment allows it
+    pub fn load<S, V>(
+        file_paths: &[Option<&Path>],
+        env_prefix: &str,
+        flag_overrides: HashMap<S, V>,
+    ) -> Result<Config>
+    where
+        S: AsRef<str>,
+        V: Into<Value>,
+    {
+        let mut builder = config::Config::builder();
+
+        // layer on config options from files
+        for path in file_paths.iter().flatten() {
+            if path.exists() {
+                let p = path.to_str().ok_or_else(|| anyhow::anyhow!("empty path"))?;
+                builder = builder.add_source(File::with_name(p));
+            }
+        }
+
+        // next, add any environment variables
+        builder = builder.add_source(
+            Environment::with_prefix(env_prefix)
+                .separator("__")
+                .try_parsing(true),
+        );
+
+        // finally, override any values
+        for (flag, val) in flag_overrides.into_iter() {
+            builder = builder.set_override(flag, val)?;
+        }
+
+        let cfg = builder.build()?;
+        debug!("make_config:\n{:#?}\n", cfg);
+        let cfg = cfg.try_deserialize()?;
+        Ok(cfg)
+    }
+
+    /// Constructs a `DerpMap` based on the current configuration.
+    pub fn derp_map(&self) -> Option<DerpMap> {
+        if self.derp_regions.is_empty() {
+            return None;
+        }
+
+        let mut regions = HashMap::new();
+        for region in &self.derp_regions {
+            regions.insert(region.region_id, region.clone());
+        }
+
+        Some(DerpMap { regions })
+    }
+}
+
+fn default_derp_region() -> DerpRegion {
+    DerpRegion {
+        region_id: 1,
+        nodes: vec![DerpNode {
+            name: "default-1".into(),
+            region_id: 1,
+            host_name: "derp.iroh.computer".into(),
+            stun_only: false,
+            stun_port: 3478,
+            ipv4: UseIpv4::Some("35.175.99.113".parse().unwrap()),
```

Comment:
move this IP to a constant & add a comment?

Reply:
this is config is the constant

Reply:
sure, then can you add a comment clarifying these are run by n0 / iroh maintainers? Just something so folks have a sense that this is provisioned-but-default infra